hypothesis-python/setup.py

```diff
@@ -23,6 +23,14 @@
 
 import setuptools
 
+if sys.version_info[:2] < (3, 5):
+    raise Exception(
+        "This version of Python is too old to install new versions of Hypothesis.  "
+        "Update `pip` and `setuptools`, try again, and you will automatically "
+        "get the latest compatible version of Hypothesis instead.  "
```

Comment:
Should we maybe add "or pin your installs to any version earlier than 5.0" or something like that so people don't whine at us about how it's impossible for them to update pip and setuptools because blah blah blah vendor packaging and we should fix this on our end?

Reply:
:man_shrugging: IMO anyone who can't keep their packaging toolchain up to date shouldn't be installing things from PyPI, and I'm happy to make that point if they turn up on our issue tracker.  ("If you can `python -m pip install hypothesis`, you can darn well `python -m pip install -U pip`")

Mostly though I expect that to be very rare, don't want to suggest people pin to old versions because they end up telling their friends about things "Hypothesis can't do" (on some ancient version), and we can always add to this warning later.